SPR-10502 - PropertyPlaceholder multiple locations resolution

fdesu opened the pull request:
This commit fixes an issue, when in XML defined config there is context:property-resolver with
the value if form of value="${placeholder}" thus it should be resolved with system property. Issue occurs when property already resolved, but appropriate splitting by comma already had passed.

Issue: https://jira.spring.io/browse/SPR-10502

fdesu commented:
Please review that simple fix.
I have also commented appropriate JIRA page.

fdesu commented:
@snicoll Can you please review and maybe push further this PR?
Thank you in advance.

snicoll commented:
I am not sure I like that change. I was about to try to resolve the placeholders in the parser itself rather but it looks like the editor may also resolve the array when translating the string to a `Resource`. I am sure @jhoeller has an opinion about the best course of action.

fdesu commented:
Thanks a lot for your opinion @snicoll. Agreed, with you, but, yes, let's just wait for an opinion from @jhoeller

fdesu commented:
@snicoll 
I've updated PR with your suggestions, and added some more tests to cover changes.
Could you please take a look?

fdesu committed:
Changes enable fix for issue, when in XML defined
config there is <context:property-resolver> with
the value in form of value="${placeholder}" thus
it should be resolved with system property.

Resolution of that property should be on early stage
when BeanDefinition creation still on the Parsing
stage.

Issue: SPR-10502

fdesu commented:
@jhoeller resolved the original issue with more extensive changes
